packages/app/obojobo-document-engine/src/scripts/oboeditor/components/hovering-preview.js

```diff
@@ -0,0 +1,70 @@
+import React, { useRef, useLayoutEffect, useMemo } from 'react'
+import { ReactEditor, useSlate } from 'slate-react'
+import { Editor } from 'slate'
+import katex from 'katex'
+
+import './hovering-preview.scss'
+const LATEX_MARK = '_latex'
+
+const HoveringPreview = ({ pageEditorContainerRef }) => {
+	const ref = useRef()
+	const editor = useSlate()
+	const [leaf] = editor.selection
+		? Editor.leaf(editor, editor.selection, { edge: 'start' })
+		: [{ text: '' }, []]
+
+	// SHOW, HIDE, AND UPDATE LOCATION ON SCREEN
+	// only runs when text changes OR window width changes
+	useLayoutEffect(() => {
+		// bail if nothing selected
+		const el = ref.current
+		if (!el) return
+
+		// bail if editor isnt selecting a latex mark
+		if (!editor.selection || !ReactEditor.isFocused(editor) || !leaf[LATEX_MARK]) {
+			el.removeAttribute('style') // remove any styles we created
+			return
+		}
+
+		// locate the dom's selected element's parent
+		const domSelection = window.getSelection()
+		const domRange = domSelection.getRangeAt(0)
+		const parent = domRange.commonAncestorContainer.parentNode
+
+		// If the parent is not a span, we have selected across multiple pieces of text
+		// so we should hide the preview to prevent chaos as multiple previews pop up
+		if (parent.tagName.toLowerCase() !== 'span') {
+			el.removeAttribute('style')
+			return
+		}
+
+		// Calculate styles to set display location
+		const pageEditorRect = pageEditorContainerRef.current.getBoundingClientRect()
+		const rect = parent.getBoundingClientRect()
+
+		el.style.opacity = 1
+		el.style.top = `${rect.top - pageEditorRect.top - el.offsetHeight - 6}px`
+		el.style.left = `${rect.left - pageEditorRect.left - el.offsetWidth / 2 + rect.width / 2}px`
+	}, [window.innerWidth, leaf.text, leaf])
```

Comment:
Added conditional watchers to useLayoutEffect - this should now only render when one of these changes

* window.innerWidth - no need to re-render on height changes as they have no effect on our calcs
* leaf.text - any time the leaf node text changes - this can be more often then desired because whatever is selected will feed it's own text in, latex or not
* leaf - if the leaf object itself is different (needed to recalc the location for 2 different equations w/ the same text)